frontend-next/src/pages/index.tsx

```diff
@@ -1,17 +1,99 @@
 import useTranscripts from "@/hooks/useTranscripts";
-import QueueTable from "@/components/queueTable/QueueTable";
+import QueueTable, { TableStructure } from "@/components/queueTable/QueueTable";
+import { useCallback, useMemo, useState } from "react";
+import { getCount } from "@/utils";
+import { Transcript } from "../../types";
+import { useSession } from "next-auth/react";
+import { useRouter } from "next/router";
 
 export default function Home() {
-  const { data, isLoading, isRefetching, isError, refetch } =
-    useTranscripts().transcripts;
+  const { data: session, status } = useSession();
+  const router = useRouter();
+  const { transcripts, claimTranscript } = useTranscripts();
+  const { data, isLoading, isRefetching, isError, refetch } = transcripts;
+
+  const [claimState, setClaimState] = useState({
+    claim: claimTranscript,
+    rowIndex: -1,
+  });
+
+  const handleClaim = useCallback(
+    (idx: number, transcriptId: number) => {
+      setClaimState((prev) => ({ ...prev, rowIndex: idx }));
+      if (status === "loading") {
+        alert("Authenticating.... please wait.");
+        return;
+      } else if (status === "unauthenticated") {
+        alert("You have to login to claim a transcript");
```

Comment:
shouldn't we just redirect the user to the sign-in callback `https://review.btctranscripts.com/api/auth/signin?error=Callback` instead. 

nit:  i've noticed that after this alert is displayed, the claim button spins continuously until the page is refreshed. wonder if that's working as intended.

Reply:
Mmm, you're right. That flow would be a better UX. 

@nassersaazi I noticed when users claim transcripts, the transcripts are still on the transcription queue. Is that meant to be so? @adamjonas I would love your input here. Perhaps I use the status to determine whether the transcript would have the claim action or not. 

Yes, Jonas had the same issue. I'm working on it

Reply:
When they are claimed they are removed from the index page.

Reply:
added @nassersaazi suggestion in [this commit](https://github.com/adamjonas/transcription-queue/pull/20/commits/5d5847c11646492f4be4d2275b1365c8d19ca5e4)